### ISSUE TITLE
Wrong orientation processing for 4/7

### ISSUE THREAD

**Issue by bigcat88** [strukturag/libheif#1026]
Discovered during testing when I was trying to add the ability to write the HEIF orientation for Pillow-Heif.

Can be reproduced using heif-enc, attached two samples.

[test_png.zip](https://github.com/strukturag/libheif/files/13250611/test_png.zip)

**Comment by bigcat88** [comment]
```
    case heif_orientation_flip_vertically:
      mirror = heif_transform_mirror_direction_horizontal;
      
        case heif_orientation_rotate_90_cw_then_flip_vertically:
      rotation_ccw = 270;
      mirror = heif_transform_mirror_direction_horizontal;
      has_mirror = true;
      break;
```

**Comment by farindk** [comment]
> oh no, I didn't make it in time for the next release... ,(

5 minutes ...

**Comment by bigcat88** [comment]
rerelease if it is possible? there is just 2 typo errors in code..

**Comment by farindk** [comment]
I'm trying to reproduce this. Any hints how you found this?

**Comment by bigcat88** [comment]
> https://github.com/strukturag/libheif/blob/760d8a8ba0ad4640386beec8db0606454f197fdc/libheif/file.cc#L1015-L1029
> 
> oh no, I didn't make it in time for the next release... ,(

lines  1016, 1029

**Comment by bigcat88** [comment]
to reproduce it I attached 2 images, that produce wrong output with heif-enc

**Comment by farindk** [comment]
I can reproduce the bug when I manually set the orientation in `heif_enc.cc`. With the input files, I cannot reproduce it because they are PNG and in PNG, the Exif orientations are only informal and not used to change the orientation of the HEIC image.

**Comment by farindk** [comment]
Now ok? Can I do another release ? ;-)

**Comment by bigcat88** [comment]
Yes, thanks.

p.s.: macOS rotates PNG based on EXIF data...

**Comment by farindk** [comment]
> p.s.: macOS rotates PNG based on EXIF data...

This is such a mess ...

I tried to find any spec for the PNG 'eXIf' that specifies whether the orientation is informal only. I am pretty sure that I search for this when I implemented it, but unfortunately, I didn't write down the reference in the source.

**Comment by kmilos** [comment]
[The PNG spec says decoders might ignore Exif altogether](https://w3c.github.io/PNG-spec/#4Concepts.AncillInfo), although it is the only source of orientation info for a PNG AFAIK.

**Comment by farindk** [comment]
Yes, the small 'e' in `eXIf` means that it is non-essential and the small `f` means: safe-to-copy when unknown to editors.

ICC data in PNG (`iCCP`) is also considered non-essential. From that I conclude that non-essential does not mean that the output is completely the expected outcome but something that is at least usable. That would mean that non-essential does not mean that we should ignore the EXIF orientation.

However, safe-to-copy for Exif only makes sense together with the following paragraph from the [PNG specs](https://w3c.github.io/PNG-spec/#eXIf):

> 11.3.4.5
> ...
> The eXIf chunk contains metadata concerning the original image data. If the image has been edited subsequent to creation of the Exif profile, this data might no longer apply to the PNG image data. It is recommended that unless a decoder has independent knowledge of the validity of the Exif data, the data should be considered to be of historical value only. It is beyond the scope of this specification to resolve potential conflicts between data in the eXIf chunk and in other PNG chunks.

From this, I conclude, that all data in the EXIF should be considered informational only since the image itself may have changed respective to the original.
